python_modules/dagster/dagster/_core/definitions/declarative_automation/operators/boolean_operators.py

```diff
@@ -191,3 +274,41 @@ def replace(
             if old in [self, self.get_label()]
             else copy(self, operand=self.operand.replace(old, new))
         )
+
+    @public
+    def allow(self, selection: "AssetSelection") -> "NotAutomationCondition":
+        """Applies the ``.allow()`` method across all sub-conditions.
+
+        This impacts any dep-related sub-conditions.
+
+        Args:
+            selection (AssetSelection): The selection to allow.
+        """
+        from dagster._core.definitions.asset_selection import AssetSelection
+
+        check.inst_param(selection, "selection", AssetSelection)
+        return copy(
+            self,
+            operand=self.operand.allow(selection)
+            if hasattr(self.operand, "allow")
+            else self.operand,
+        )
+
+    @public
+    def ignore(self, selection: "AssetSelection") -> "NotAutomationCondition":
+        """Applies the ``.ignore()`` method across all sub-conditions.
+
+        This impacts any dep-related sub-conditions.
+
+        Args:
+            selection (AssetSelection): The selection to ignore.
+        """
+        from dagster._core.definitions.asset_selection import AssetSelection
```

Comment:
I haven't thought too deeply on this, but is there an opportunity to reduce the amount of boilerplate here?

Maybe some combination of a generic function to replace the children of a given condition or a function that factors out the "cond.ignore(selection) if hasattr(cond, "ignore") else cond" bit?

Reply:
got it -- I agree that it's not obviously better to do it that way, and we can always refactor later if we decide that's worth it.